drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c

```diff
@@ -141,12 +142,10 @@ static int eth_nxp_enet_qos_tx(const struct device *dev, struct net_pkt *pkt)
 	return 0;
 }
 
-static void tx_dma_done(struct k_work *work)
+static void tx_dma_done(const struct device *dev)
 {
-	struct nxp_enet_qos_tx_data *tx_data =
```

Comment:
This change is avoiding using a workq to do packet reference freeing in the ISR. It is unfortunate that we can't use the workq for something that is truly a background low priority activity.